### PR TITLE
Feature/sof 4712 - Bandgap binning

### PR DESCRIPTION
**Pull request by AcylSilane**

### CODE REVIEW

File: DigitalEcosystem/exploratory/exfoliation_bandgap_binning/exfoliation_bandgap_binning.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+atoms_col = ['atoms_object (unitless)']
+
+
+# In[]:
+
+
+noble_gases = ['He', 'Ne', 'Ar', 'Kr', 'Xe', 'Rn']
+fblock = ['La', 'Ce', 'Pr', 'Nd', 'Pm', 'Sm', 'Eu', 'Gd', 'Tb', 'Dy', 'Ho', 'Er', 'Tm', 'Yb', 'Lu',
+          'Ac', 'Th', 'Pa', 'U',  'Np', 'Pu', 'Am', 'Cm', 'Bk', 'Cf', 'Es', 'Fm', 'Md', 'No', 'Lr']
+d_synths = ['Rf', 'Db', 'Sg', 'Bh', 'HS', 'Mt', 'Ds', 'Rg', 'Cn', 'Nh', 'Fl', 'Mc', 'Lv', 'Ts', 'Og']
```
Comment (timurbazhirov) [inline DigitalEcosystem/exploratory/exfoliation_bandgap_binning/exfoliation_bandgap_binning.py]: These definitions should go into some central place and be reused across notebooks

File: DigitalEcosystem/exploratory/exfoliation_bandgap_binning/exfoliation_bandgap_binning.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+test_x_reg = np.nan_to_num(test[xenonpy_matminer_descriptors + ['low_bandgap']].to_numpy())
+test_y_reg = np.nan_to_num(test[target_column].to_numpy())
+
+
+# In[]:
+
+
+current_reg = None
+best_reg = None
+def keep_best_reg(study, trial):
```
Comment (timurbazhirov) [inline DigitalEcosystem/exploratory/exfoliation_bandgap_binning/exfoliation_bandgap_binning.py]: Again:

- no shornames
- add explanation

File: DigitalEcosystem/exploratory/exfoliation_bandgap_binning/exfoliation_bandgap_binning.py (hunk 1 above)
Comment (AcylSilane) [reply DigitalEcosystem/exploratory/exfoliation_bandgap_binning/exfoliation_bandgap_binning.py]: Moved

File: DigitalEcosystem/exploratory/exfoliation_bandgap_binning/exfoliation_bandgap_binning.py (hunk 2 above)
Comment (AcylSilane) [reply DigitalEcosystem/exploratory/exfoliation_bandgap_binning/exfoliation_bandgap_binning.py]: Expanded the names and added comments

**Comment by AcylSilane** [comment]
Merging per https://exabyte.atlassian.net/browse/SOF-4712?focusedCommentId=42237

**Pull request merged**